### PR TITLE
v4.1.x: oshmem: remove unnecessary dependencies to ORTE

### PR DESCRIPTION
**Pull request by hppritcha**
either use OPAL or OMPI layers, since ORTE layer
is not present when PMIx RTE is used

related to #8106

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>
(cherry picked from commit 10cb9f6f9ea4c32f0efcda1b96606b965f8f9da0)

Conflicts:
	oshmem/info/info.c
	oshmem/mca/memheap/base/memheap_base_select.c
	oshmem/mca/spml/ikrit/spml_ikrit_component.c

### CODE REVIEW

File: oshmem/info/info.c [hunk 1]
```diff
@@ -1,6 +1,9 @@
 /*
  * Copyright (c) 2015      Mellanox Technologies, Inc.
  *                         All rights reserved.
+ * Copyright (c) 2018      Amazon.com, Inc. or its affiliates.  All Rights reserved.
+ * Copyright (c) 2019      Research Organization for Information Science
```
Comment (hoopoepg) [inline oshmem/info/info.c:5]: minor question: why 2019? shouldn't it be year of change?

**Review by hoopoepg — APPROVED** [review]

File: oshmem/info/info.c (hunk 1 above)
Comment (hoopoepg) [reply oshmem/info/info.c:5]: it is cherry-pick...
thank you for help

**Comment by hppritcha** [comment]
i don't care about whether we fix issue #8106 or not.  If we don't plan to apply these patches then I'll just close #8106 as won't fix.

**Comment by bwbarrett** [comment]
I think we should fix $8106, but this doesn't seem like the way to do it.  It looks like shmem_info isn't linking in libopen-rte in those branches, which seems like what we should fix.

**Comment by hppritcha** [comment]
hmmm... I think the issue is that the configure option the user had selected (--with-ompi-pmix-rte) is supposed to not build/link in orte.

**Comment by rhc54** [comment]
Correct - you are supposed to be able to build OMPI without ORTE in the v4.x series. Problem is that this is broken by OSHMEM at the moment as it directly references ORTE. Only other solution is to disable OSHMEM if the user configures without ORTE.

**Comment by bwbarrett** [comment]
Convert to draft until we figure out what we're going to do with this change and release branches.

**Comment by janjust** [comment]
Closing due to concerns over the potential side effects this may bring into release branches. Issue #8106 will remain open until further discussion.

**Pull request closed without merging**